data-lineage/src/main/java/com/wipro/ats/bdre/lineage/LineageMain.java

```diff
@@ -69,16 +71,28 @@
 	private Integer processId;
 	private Long instanceId;
 	private String defaultHiveDbName;
+	private LineageQuery lineageQuery;
 
 	//	private int nextId = 0;
 	private int subquerySeq = 0;
 	private PersistenceUnit persistenceUnit;
 
-	public static void main(String[] args) throws IOException, ParseException,
+	/*public LineageMain(String query, Integer processId, Long instanceId, String defaultHiveDbName) {
+		this.query = query;
+		this.processId = processId;
+		this.instanceId = instanceId;
+		this.defaultHiveDbName = defaultHiveDbName;
+	}*/
+	public static void lineageMain(LineageQuery lineageQuery,String args1, String args2, String args3) throws IOException, ParseException,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, org.apache.hadoop.hive.ql.parse.ParseException, org.apache.hadoop.hive.ql.parse.SemanticException, java.lang.Exception [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1160)